### DISCUSSION TITLE
Docker Host Network Deployment Issues...

### DISCUSSION THREAD

**Discussion by stuu3k** [Lissy93/dashy#477] [Help and Support]
Hi Lissy,

I don't know how I managed to get Host Networking working previously, I tried deploying Dashy for a mate yesterday and I could only get it to work using Bridge Network which seems to give us some limitations in this config.

I then decided to update mine today and see where I went wrong and IDK if somethings changed or what but it keeps failing, the logs seem to compain about the HostName?

This is the command I am using to install, aside from some data being hidden...

docker run -d \
  --dns 10.0.0.2 \
  --dns 10.0.0.20 \
  --name Dashy \
  --hostname Dashy \
  --network host \
  --restart always \
  -v /xxxx/Dashy/conf.yml:/app/public/conf.yml \
  -v /xxxx/Dashy/Icons:/app/public/item-icons  \
  -e PORT=8019 \
  -e PUID=xxxx \
  -e PGID=100 \
  lissy93/dashy:latest

This is pointing to my existing config files which i know are good but when I start the Docker, I see the following...

yarn run v1.22.15
$ npm-run-all --parallel build-watch start
$ vue-cli-service build --watch --mode production
$ node server
Checking config file against schema...
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  Warning: 6 issues found in config file  
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
1. required must have required property 'serverUrl' in /appConfig/auth/keycloak
2. required must have required property 'realm' in /appConfig/auth/keycloak
3. required must have required property 'clientId' in /appConfig/auth/keycloak
4. maximum must be <= 5 in /sections/0/displayData/rows
5. maximum must be <= 5 in /sections/6/displayData/rows
6. maximum must be <= 5 in /sections/10/displayData/cols
SSL Not Enabled: Public key not present
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^
Error: getaddrinfo ENOTFOUND Dashy
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'Dashy'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "start" exited with 1.
error Command failed with exit code 1.
=========================================================

Is this a DNS Reverse lookup issue? It's behind a Reverse Proxy so it may not always respond with Dashy is it checking that and failing or something else?!

Thanks

**Comment by lissy93** [comment]
So there have been some big changes to the Dockerfile recently, but that doesn't seem to be the cause as otherwise the app wouldn't have got this far. This looks more like a DNS error 🤔 

You're right that it could be where the app's hostname + port is printed, although none of that code has been changed in months. But I can wrap add some checks in place to provide a safe fallback if there's an error while fetching that info.

I'll test and push those changes to the `latest` image now, and we'll see how that gets on.

**Reply by lissy93** [reply]
Update, see PR #480
Update 2, Merged.

**Comment by stuu3k** [comment]
Hi Lissy,

Thanks, that did it, I have found this now works providing one of the following is true...

1. You do not specify the "hostname" field at all.
2. If you do Specify it the "hostname" field it must be right.

That's fine I have update the deployment scheduled task on both boxes and repulled the image all working well, this is the code I am using if anyone is interested...

docker run -d \
  --dns DNS_SERVER_IP_1 \
  --dns DNS_SERVER_IP_2 \
  --name Dashy \
  --hostname HOST_BOXES_HOSTNAME \
  --domainname DOMAIN.XYZ \
  --network host \
  --restart always \
  -v /volume1/docker/Dashy/conf.yml:/app/public/conf.yml \
  -v /volume1/docker/Dashy/Icons:/app/public/item-icons  \
  -e PORT=8019 \
  -e PUID=**** \
  -e PGID=100 \
  lissy93/dashy:latest

DNS Servers, Hostname and Domainname lines can be removed if not wanted.
PUID and PGID Should be set on Synology boxes to ensure access to files \ folders on drive.

Thanks
